Lib/test/test_asyncio/test_eager_task_factory.py

```diff
@@ -0,0 +1,241 @@
+"""Tests for base_events.py"""
+
+import gc
+import time
+import unittest
+from types import GenericAlias
+from unittest import mock
+
+import asyncio
+from asyncio import base_events
+from asyncio import tasks
+from test.test_asyncio import utils as test_utils
+from test.test_asyncio.test_tasks import get_innermost_context
+from test import support
+
+MOCK_ANY = mock.ANY
+
+
+def tearDownModule():
+    asyncio.set_event_loop_policy(None)
+
+
+class EagerTaskFactoryLoopTests(test_utils.TestCase):
+
+    def new_task(self, loop, coro, name='TestTask', context=None):
+        return tasks.Task(coro, loop=loop, name=name, context=context)
+
+    def new_future(self, loop):
+        return asyncio.Future(loop=loop)
+
+    def setUp(self):
+        super().setUp()
+        self.loop = asyncio.new_event_loop()
+        self.loop.set_task_factory(asyncio.eager_task_factory)
+        self.set_event_loop(self.loop)
+
+    def test_eager_task_factory_set(self):
+        self.assertIs(self.loop.get_task_factory(), asyncio.eager_task_factory)
+
+    def test_await_future_during_eager_step(self):
+
+        async def set_result(fut, val):
+            fut.set_result(val)
+
+        async def run():
+            fut = self.loop.create_future()
+            t = self.loop.create_task(set_result(fut, 'my message'))
+            # assert the eager step completed the task
+            self.assertTrue(t.done())
+            return await fut
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result,'my message')
```

Comment:
I'm not sure what this is testing beyond the `test_eager_completion()` below? This I wouldn't expect `Future`s to behave differently in the case of eager execution.

Reply:
I wouldn't expect Future to behave differently - so let's test that that's indeed the case.

I'm not 100% sure this is testing exactly what @gvanrossum asked me about "awaiting a future during the eager step", so hopefully he can chime in here.